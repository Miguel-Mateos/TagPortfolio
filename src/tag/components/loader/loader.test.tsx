import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Loader, {ILoaderProps} from './loader';

const LoaderExample = (props: ILoaderProps) => (
  <Loader
    percentage={props.percentage}
    className="tag-ds"
    data-testid="loader-test"
    spinner={props.spinner}
    title={props.title}
    subtitle={props.subtitle}
    success={props.success}
    error={props.error}
  />
);

test('render Loader and match snap', () => {
  const component = renderer.create(<LoaderExample />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('loader small', () => {
  const {getByTestId} = render(<LoaderExample percentage={{value: 50, show: false}} />);
  const loader = getByTestId('loader-test-svg-container');
  expect(loader).toHaveClass('svg-container_circular');
});

test.skip('correct interval load', () => {
  const {getByTestId} = render(<LoaderExample />);

  const loadingProgress = getByTestId('loader-test-progress');

  let currentProgress = 0;
  const timer = setInterval(() => {
    if (currentProgress === 100) clearInterval(timer);
    expect(loadingProgress).toHaveClass(`loader-background_${currentProgress}`);
    currentProgress += 25;
  }, 2000);

  expect(currentProgress).toBeLessThanOrEqual(100);
});

test('render loader automatically', () => {
  const {getByTestId} = render(<LoaderExample title="Test Title" subtitle="Test subtitle" automatic />);
  const loader = getByTestId('loader-test-svg-container');
  setTimeout(() => {
    expect(loader).toHaveClass('success');
  }, 5000);
});

test('render title and subtitle', () => {
  const {getByTestId} = render(<LoaderExample title="Test Title" subtitle="Test subtitle" />);

  const title = getByTestId('loader-test-title');
  const subtitle = getByTestId('loader-test-subtitle');
  expect(title).toHaveTextContent('Test Title');
  expect(subtitle).toHaveTextContent('Test subtitle');
});

describe('finish loading both cases', () => {
  test('success case', () => {
    const {getByTestId} = render(
      <LoaderExample title="Test Title" subtitle="Test subtitle" percentage={{value: 100, show: true}} />
    );
    setTimeout(() => {
      const loader = getByTestId('loader-test-svg-container');
      expect(loader).toHaveClass('success');
    }, 2000);
  });

  test('error case', () => {
    const {getByTestId} = render(
      <LoaderExample title="Test Title" error subtitle="Test subtitle" percentage={{value: 100, show: true}} />
    );
    setTimeout(() => {
      const loader = getByTestId('loader-test-svg-container');
      expect(loader).toHaveClass('error');
    }, 2000);
  });
});

test('render spinner', () => {
  const {getByTestId} = render(<LoaderExample spinner />);
  const loader = getByTestId('loader-test-spinner');
  screen.debug();
  expect(loader).toHaveClass('spinner');
});
