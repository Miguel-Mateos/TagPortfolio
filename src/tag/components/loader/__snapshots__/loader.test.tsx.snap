// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Loader and match snap 1`] = `
<div
  className="loader-wrapper tag-ds"
  data-testid="loader-test"
>
  <div
    className="svg-container "
    data-percentage="0%"
    data-testid="loader-test-svg-container"
  >
    <svg
      className="svg-content"
      fill="none"
      height="32"
      viewBox="0 0 56 32"
      width="56"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="oval-svg"
        d="M28 2H40C47.732 2 54 8.26801 54 16C54 23.732 47.732 30 40 30H28H16C8.26801 30 2 23.732 2 16C2 8.26801 8.26801 2 16 2H28Z"
        stroke="rgb(0, 154, 204)"
        strokeLinecap="round"
        strokeWidth="4"
      />
    </svg>
  </div>
</div>
`;
